tests/experimental/datasources/test_viral_snippets.py

```diff
@@ -157,10 +158,68 @@ def test_zep_simple_validate_workflow(zep_file_context: FileDataContext):
     assert result["success"] is True
 
 
-@pytest.mark.xfail(reason="TypeError: cannot pickle 'module' object")
-def test_save_datacontext(zep_file_context: FileDataContext):
+def test_save_project_does_not_break(zep_file_context: FileDataContext):
+    print(zep_file_context.zep_config)
+    zep_file_context._save_project_config()
+
+
+def test_variables_save_config_does_not_break(zep_file_context: FileDataContext):
+    print(zep_file_context.zep_config)
+    print(zep_file_context.variables)
     zep_file_context.variables.save_config()
 
 
+def test_save_datacontext_persists_zep_config(
+    file_dc_config_dir_init: pathlib.Path, zep_only_config: GxConfig
+):
+    config_file = file_dc_config_dir_init / FileDataContext.GX_YML
+
+    initial_yaml = config_file.read_text()
+    for ds_name in zep_only_config.datasources:
+        assert ds_name not in initial_yaml
+
+    context: FileDataContext = get_context(
+        context_root_dir=config_file.parent, cloud_mode=False
+    )
+
+    context.zep_config = zep_only_config
+    context._save_project_config()
+
+    final_yaml = config_file.read_text()
+    diff = difflib.ndiff(initial_yaml.splitlines(), final_yaml.splitlines())
```

Comment:
I'm not sure generating this `diff` is worthwhile if it can't be asserted on. I think your assert below is sufficient. You could also turn the yaml into a dict and assert keys there.

Reply:
I had left it in to help debuging in the event the test fails.
But I can remove it.